packages/web/src/store/slices/package.ts

```diff
@@ -0,0 +1,61 @@
+import { createAsyncThunk, createSlice, SerializedError } from '@reduxjs/toolkit';
+import { client, GetPackageInfoOutput } from '../../services/apiClient';
+
+export type PackageInfoState = {
+  isLoading: boolean;
+  error?: SerializedError;
+  packageInfo?: GetPackageInfoOutput;
+};
+
+export type PackageInfoSlice = Record<string, PackageInfoState>;
+
+export const requestPackageInfo = createAsyncThunk(
+  'package/getInfo',
+  async ({ packageName }: { packageName: string }) => {
+    return client.query('getPackageInfo', { packageName });
+  }
+);
+
+const packageInfo = createSlice({
+  name: 'package',
+  initialState: {} as PackageInfoSlice,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(requestPackageInfo.pending, (state, action) => {
+        const { packageName } = action.meta.arg;
+        const previousPackageState = state[packageName] ?? null;
+
+        state[packageName] = {
+          ...previousPackageState,
```

Comment:
nit: We're fully re-writing all of the fields here, so the spread is meaningless at the moment. Up to you whether we want to keep it.